### PR TITLE
feature: Webpack Spring Cleaning

### PR DESCRIPTION
**Pull request by icirellik**
This cleans up the webpack configurations by deduplicating a lot of the
shared configurations.

### CODE REVIEW

File: webpack/envs/novoCommonConfig.js
Comment (joeyAghion) [inline webpack/envs/novoCommonConfig.js:23]: Still relevant?

**Review by joeyAghion — APPROVED** [review]
Note that there are some test failures at the moment.

Anything we should review or QA with particular care?

**Comment by damassi** [comment]
Before this gets merged I want to give it a thorough review

File: webpack/envs/clientCommonConfig.js
Comment (damassi) [inline webpack/envs/clientCommonConfig.js:12]: Can we rename this `webpackEnv`? With `env` being used extensively throughout, `commonEnv` / `env` is confusing.

File: webpack/envs/commonEnv.js
Comment (damassi) [inline webpack/envs/commonEnv.js:7]: Do we envision there ever being anything other than "standard"? Until that needs to be addressed, could we go with something like
- `webpackEnv.devTool
- `webpackEnv.stats`
- `webpackEnv.minimizer`
- `webpackEnv.resolve`

So as to avoid adding overloaded terms.

File: webpack/envs/novoCommonConfig.js
Comment (icirellik) [reply webpack/envs/novoCommonConfig.js:23]: Nah, it was useful to show that nearly 50% of our bundle source is duplicated GraphQL queries. I wouldn't use this in production.

**Review by damassi — APPROVED** [review]
Two suggested changes, both non-blocking 👍

File: webpack/envs/clientCommonConfig.js
Comment (icirellik) [reply webpack/envs/clientCommonConfig.js:12]: I'll think on it and see if I can come up with a more suitable name. It's unlikely that naming will be the main point of confusion for those working with Webpack.

File: webpack/envs/commonEnv.js
Comment (icirellik) [reply webpack/envs/commonEnv.js:7]: I went back and forth on this, I really didn't want to overload the base terms. Quite literally the ones you are suggesting be overloaded

I think it would be ok to separate out prod and dev, I don't expect those to stay in sync long term anyway.

File: webpack/envs/clientCommonConfig.js
Comment (damassi) [reply webpack/envs/clientCommonConfig.js:12]: If there's `env` and `commonEnv` side by side, that's def confusing. But if `commonEnv` actually refers to `webpackEnv` then problem solved.

**Pull request merged**
